timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -553,32 +553,42 @@ void __global__ k_compute_nonbonded_target_atom_energies(
 }
 
 // NUM_BLOCKS is the number of blocks that k_compute_nonbonded_target_atom_energies is run with. Decides the number
-// of values that need to be accumulated per atom.
+// of values that need to be accumulated per atom. Also dictates the number of threads
 template <typename RealType, int NUM_BLOCKS>
 void __global__ k_accumulate_atom_energies_to_per_mol_energies(
     const int target_mols,
     const int *__restrict__ mol_offsets,            // [target_mols + 1]
     const __int128 *__restrict__ per_atom_energies, // [target_atoms, NUM_BLOCKS]
     __int128 *__restrict__ per_mol_energies) {
-    const int idx = blockIdx.x * blockDim.x + threadIdx.x;
 
-    if (idx >= target_mols) {
-        return;
-    }
+    int mol_idx = blockIdx.x;
 
-    __int128 energy_accumulator = 0;
+    static_assert(NUM_BLOCKS <= 256 && (NUM_BLOCKS & (NUM_BLOCKS - 1)) == 0);
+    volatile __shared__ __int128 block_energy_buffer[NUM_BLOCKS];
```

Comment:
Is `volatile` required for the `block_energy_reduce` call? If so, the usage is inconsistent, many other kernels that call into it do not declare `volatile` smem.

Reply:
Should be consistent, and use `volatile`

Reply:
Made consistent in https://github.com/proteneer/timemachine/pull/1283/commits/1248de98c89792cb6afa298d5b91576fd220c518

Reply:
Fixed up issue with volatile + atomics in https://github.com/proteneer/timemachine/pull/1283/commits/eda228f686fd00bfd352d3e4cf0a0edf46cb7534

Reply:
Removed all volatiles in https://github.com/proteneer/timemachine/pull/1283/commits/3933dbaf46f5984e2b97077a2a60140cd4f56992 since we have `__syncthreads` everywhere to ensure values are synced.